### PR TITLE
Master dev

### PR DESCRIPTION
**Pull request by alexk2111**

### CODE REVIEW

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+               val amount = editTextUserName.text.toString()
+               val action = LoginFragmentDirections.actionLoginFragmentToResultFragment(amount)
+               findNavController().navigate(action)
+           }
+       }
+    }
+
+    private fun checkUserName(userName: String): Boolean {
+        if (userName.isEmpty() || userName.contains( " ")) {
+            binding.editTextUserName.error = "Field cannot be empty or or contain spaces"
```
Comment (shadwork) [inline app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Text should be in android resources

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import com.onix.okucherenko.loginapplication.databinding.FragmentLoginBinding
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [LoginFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class LoginFragment : Fragment() {
+
+    private var _binding: FragmentLoginBinding? = null
```
Comment (shadwork) [inline app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Ok, binding is fine but also fragment should use ViewModel https://developer.android.com/codelabs/kotlin-android-training-view-model#0

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    private val binding get() = _binding!!
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
```
Comment (shadwork) [inline app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Why actually this code needed

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    private fun checkUserName(userName: String): Boolean {
+        if (userName.isEmpty() || userName.contains( " ")) {
+            binding.editTextUserName.error = "Field cannot be empty or or contain spaces"
+            return false
+        }
+        return true
+    }
+
+    private fun checkPassword(password: String): Boolean {
+        if (password.uppercase() == password) {
```
Comment (shadwork) [inline app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Should be 'when' statement

File: app/src/main/java/com/onix/okucherenko/loginapplication/ResultFragment.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.fragment.navArgs
+import com.onix.okucherenko.loginapplication.databinding.FragmentResultBinding
+
+
```
Comment (shadwork) [inline app/src/main/java/com/onix/okucherenko/loginapplication/ResultFragment.kt]: Add ViewModel please

File: app/src/main/res/layout/fragment_login.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (shadwork) [inline app/src/main/res/layout/fragment_login.xml]: Should be ConstraintLayout

File: app/src/main/res/layout/fragment_splash.xml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (shadwork) [inline app/src/main/res/layout/fragment_splash.xml]: Use constraint layout

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1,3 +1,11 @@
 <resources>
     <string name="app_name">LoginApplication</string>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (shadwork) [inline app/src/main/res/values/strings.xml]: Remove it

**Review by shadwork — CHANGES_REQUESTED** [review]
Отлично, почитай про ViewModel и заимплементь вместе с байндингом

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt (hunk 1 above)
Comment (alexk2111) [reply app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Done

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (alexk2111) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt (hunk 4 above)
Comment (alexk2111) [reply app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Done

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt (hunk 3 above)
Comment (alexk2111) [reply app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Done

File: app/src/main/res/layout/fragment_splash.xml (hunk 7 above)
Comment (alexk2111) [reply app/src/main/res/layout/fragment_splash.xml]: Done

File: app/src/main/res/layout/fragment_login.xml (hunk 6 above)
Comment (alexk2111) [reply app/src/main/res/layout/fragment_login.xml]: Done

File: app/src/main/java/com/onix/okucherenko/loginapplication/ResultFragment.kt (hunk 5 above)
Comment (alexk2111) [reply app/src/main/java/com/onix/okucherenko/loginapplication/ResultFragment.kt]: Done

File: app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt (hunk 2 above)
Comment (alexk2111) [reply app/src/main/java/com/onix/okucherenko/loginapplication/LoginFragment.kt]: Done

**Pull request merged**